Project/Pages/catalogPage.ts

```diff
@@ -26,12 +26,14 @@ export class CatalogPage {
 
   async applyFilter(filterID: string) {
     await this.page.locator(`label[for="${filterID}"]`).check();
+    await this.page.waitForResponse(response => response.url().includes('https://telemart.ua/ua/monitors/filter/') && response.status() === 200 && response.request().method() === 'GET', {timeout: 6000});
     await expect(this.page.locator(`label[for="${filterID}"]`)).toBeChecked();
   }
 
   async applyMultipleFilters(listOfFilters: Array<string>) {
     for (const filter of listOfFilters) {
       await this.page.locator(`label[for="${filter}"]`).check();
+      await this.page.waitForResponse(response => response.url().includes('https://telemart.ua/ua/monitors/filter/') && response.status() === 200 && response.request().method() === 'GET', {timeout: 6000});
```

Comment:
added waitForResponse to avoid flakiness during last filter selection. without checking the response filter was not applying from time to time